cdf_fabric_replicator/time_series.py

```diff
@@ -75,13 +76,24 @@ def run(self) -> None:
                 time.sleep(sleep_time)
 
     def process_subscriptions(self) -> None:
-        for subscription in self.config.subscriptions:
-            for partition in subscription.partitions:
-                with ThreadPoolExecutor() as executor:
-                    future = executor.submit(
-                        self.process_partition, subscription, partition
-                    )
-                    logging.debug(future.result())
+        if (
+            self.cognite_client.time_series.subscriptions.retrieve(
+                external_id=self.config.subscription.external_id
+            )
+            is None
+        ):
+            try:
+                self.create_subscription(self.config.subscription.num_partitions)
+            except CogniteAPIError as e:
+                logging.error(f"Error creating subscription: {e}")
+                raise e
+
+        for partition in range(0, self.config.subscription.num_partitions):
+            with ThreadPoolExecutor() as executor:
+                future = executor.submit(
+                    self.process_partition, self.config.subscription, partition
```

Comment:
It appears that a failure writing a data point to the lakehouse would take down this entire thread without logging the failure or ensuring a retry.

Reply:
See the other comments for information on threading, which has been removed. Added try / catch logic to log errors for writing to Lakehouse and iterating over timeseries. Additional error handling and logic is scheduled for implementation in the upcoming story # 37362 ( As a platform operator, I want a robust replication for the Fabric Connector ).